src/Spice86.Core/Emulator/Devices/Input/Keyboard/Keyboard.cs

```diff
@@ -46,13 +50,43 @@ public class Keyboard : DefaultIOPortHandler {
     public override byte ReadByte(int port) {
         byte? scancode = LastKeyboardInput.ScanCode;
         if (scancode == null) {
-            return 0;
+            scancode = 0;
+        }
+        
+        return port switch {
+            KeyboardPorts.IOPort => scancode.Value,
```

Comment:
Shouldnt we read the keyboard buffer just when we are requested to read scan code?

Reply:
Not really. Maybe VGA ports ?
This is the same as here:
https://github.com/sarah-walker-pcem/pcem/blob/dev/src/keyboard/keyboard_at.c#L314